lib/api/core/auth/formatProcessing.js

```diff
@@ -64,7 +64,24 @@ module.exports = {
   formatUserForSerialization: function (kuzzle, user, hydrate) {
     return kuzzle.pluginsManager.trigger('security:formatUserForSerialization', user)
       .then(triggeredUser => {
-        var response = triggeredUser;
+        var deferred = q.defer();
```

Comment:
Avoid use of `defer()` when it can be avoided. This improves readability and limits the risks of unresolved promises.

In this case, this function can be written like this:
```js
       if (typeof triggeredUser.profile === 'string') {
         return kuzzle.repositories.profile.loadProfile(triggeredUser.profile)
           .then(profile => {
             triggeredUser.profile = profile;
             return q(triggeredUser);
           });
       }

       return q(triggeredUser);
```